internal/pkg/skuba/replica/replica.go

```diff
@@ -0,0 +1,278 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package replica
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/SUSE/skuba/internal/pkg/skuba/kubernetes"
+)
+
+type affinity int
+
+const (
+	preferred affinity = iota
+	required
+)
+
+const (
+	highAvailabilitylabel = "caasp.suse.com/skuba-replica-ha"
+	minSize               = 2
+	patchAffinityRemove   = `[
+		{
+			"op":"remove",
+			"path":"/spec/template/spec/affinity"
+		}
+	]`
+	patchAffinityRequired = `{
+		"spec": {
+			"template": {
+				"spec": {
+					"affinity": {
+						"podAntiAffinity": {
+							"requiredDuringSchedulingIgnoredDuringExecution": [
+								{
+									"labelSelector": {
+										"matchExpressions": [
+											{
+												"key": "app",
+												"operator": "In",
+												"values": [
+													"%s"
+												]
+											}
+										]
+									},
+									"topologyKey": "kubernetes.io/hostname"
+								}
+							]
+						}
+					}
+				}
+			}
+		}
+	}`
+	patchAffinityPreferred = `{
+		"spec": {
+			"template": {
+				"spec": {
+					"affinity": {
+						"podAntiAffinity": {
+							"preferredDuringSchedulingIgnoredDuringExecution": [
+								{
+									"weight": 100,
+									"podAffinityTerm": {
+										"labelSelector": {
+											"matchExpressions": [
+												{
+													"key": "app",
+													"operator": "In",
+													"values": [
+														"%s"
+													]
+												}
+											]
+										},
+										"topologyKey": "kubernetes.io/hostname"
+									}
+								}
+							]
+						}
+					}
+				}
+			}
+		}
+	}`
+	patchReplicas = `{
+		"spec": {
+			"replicas": %v
+		}
+	}`
+)
+
+// Helper provide methods for replica update of deployment
+// resource that has highAvailabilitylabel.
+type Helper struct {
+	Client            clientset.Interface
+	ClusterSize       int
+	SelectDeployments appsv1.DeploymentList
+	Deployment        appsv1.Deployment
+	ReplicaSize       int
+	MinSize           int
+	isUpdated         bool
+}
+
+// NewHelper creates a helper to update deployment replicas.
+func NewHelper(client clientset.Interface) (*Helper, error) {
+	fromSelectors, err := client.AppsV1().Deployments(metav1.NamespaceSystem).List(
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=true", highAvailabilitylabel),
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	node, err := kubernetes.GetAllNodes(client)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Helper{
+		Client:            client,
+		ClusterSize:       len(node.Items),
+		MinSize:           minSize,
+		SelectDeployments: *fromSelectors,
+	}, nil
+}
+
+func (r *Helper) replaceAffinity(remove affinity, create affinity) error {
```

Comment:
```suggestion
func (r *Helper) replaceAffinity(remove affinity, create affinity) bool, error {
```

Suggest to remove `Hepler::isUpdated`, and use returned bool value to indicate if the replacement is successful.

Reply:
updated, thanks